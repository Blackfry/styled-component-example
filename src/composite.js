import React, { Component } from 'react'
import styled from 'styled-components'
import tile from './Assets/tile.jpg'
import logo from './Assets/logo.png'


export const Container = styled.div`
  width: 100%;
  height: auto;
  position: absolute;
  z-index: -1;
`

export const Image = styled.img`
  height: 100%;
  width: 100%;
`

export const LogoBanner = styled.span`
  height: 30%;
  width: 99.5%;
  opacity: .6;
  background-color: black;
  position: absolute;
  bottom: 0%;
  left: 0;
  padding: 5px 0px 5px 5px;
  z-index: 99
`


export const Logo = styled.img`
  height: 100%;
  float: left;
`

export const TitleText = styled.span`
  color: white;
  position: absolute;
  width: 97%;
  height: 100%;
  display: table;
  z-index: 9999
`

export const Text = styled.span`
  display: table-cell;
  vertical-align: middle;
  font-size: 7vh;
  text-align: left;
  padding-left: 22%;
  padding-bottom: 2%;
  @media screen and (max-width: 768px) { 
    font-size: 5vh;
  }
`

class Composite extends Component {

  render() {

    return (
      <Container>
        <Image src={tile} />
        <LogoBanner>
          <Logo src={logo} />
          <TitleText>
            <Text>Home and Away</Text>
          </TitleText>
        </LogoBanner>
      </Container> 
    )
  }
}

export default Composite
