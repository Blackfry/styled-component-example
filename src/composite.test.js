import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import { Container, Image, LogoBanner, Logo, TitleText, Text } from './composite'

describe('Test suite for App', () => {
  it('Container snapshot', () => {
    const component = renderer.create(
      <Container />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Image snapshot', () => {
    const component = renderer.create(
      <Image />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('LogoBanner snapshot', () => {
    const component = renderer.create(
      <LogoBanner />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('Logo snapshot', () => {
    const component = renderer.create(
      <Logo />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('TitleText snapshot', () => {
    const component = renderer.create(
      <TitleText />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
  
  it('Text snapshot', () => {
    const component = renderer.create(
      <Text />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})