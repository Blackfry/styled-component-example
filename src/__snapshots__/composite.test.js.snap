// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for App Container snapshot 1`] = `
<div
  className="sc-bdVaJa frGBwh"
/>
`;

exports[`Test suite for App Image snapshot 1`] = `
<img
  className="sc-bwzfXH fkGgTe"
/>
`;

exports[`Test suite for App Logo snapshot 1`] = `
<img
  className="sc-bxivhb drsGEi"
/>
`;

exports[`Test suite for App LogoBanner snapshot 1`] = `
<span
  className="sc-htpNat XlrGQ"
/>
`;

exports[`Test suite for App Text snapshot 1`] = `
<span
  className="sc-EHOje gjLLVX"
/>
`;

exports[`Test suite for App TitleText snapshot 1`] = `
<span
  className="sc-ifAKCX IymKh"
/>
`;
